concepts/observability-configurable-logging/02-agents.md

```diff
@@ -0,0 +1,56 @@
+To support a variety of backends and also to increase the likelihood that the user knows how to integrate, the agent should provide an active and feature-rich ecosystem.
+
+**Mature on K8S**
+It should be a battle-tested component which can scale-out and is already in use for big setups. It should have built-in kubernetes support and a lightweight footprint.
+
+## Candidates
+
+**OTEL-Collector**
+The most prominent technology nowadays is the `otel-collector` of the `OpenTelemetry` project. It is part of the CNCF and aims to cover all the three kind of data: logs, traces, metrics.
+It is quite new but already has a big community behind as it is covering the community of OpenCensus and OpenTracing. With that the ecosystem is already good enough and the majority of vendor-specific backends are supported (seeing the available exporters on the contrib repo) being based on a new vendor-neutral protocol.
+It is most probably the only real vendor-neutral project covering really all three aspects of logs, traces and metrics.
+Bigger companies are actively planning to use it at least for the trace and metrics aspects. Here, the configuration specification is still not in a stable state. For logging, the whole feature is still in a beta stadium.
+See also [more detailed here](../observability-opentelemetry/README.md)
+
+**Logs - Fluent-Bit**
+It is the traditional log collector used in Kyma already, being part of CNCF and widely adopted on K8S. It is vendor-neutral and supports a variety of generic outputs being capable of integrating with quite some backends. If a backend is not supported, you can use the fluent specific Forward protocol to connect a FluentD providing a even more richer ecosystem. It will scale-out perfect by running on every node with a minimalistic resource footprint.
+
+**Logs - FluentD**
+Written in Ruby and being more resource extensive then fluent-bit, FluentD presents a valid alternative to fluent-bit. However, it is usually recommended as the central application not running on every node but doing the central aggregation and shipment to the backend.
+
+**Logs - rsyslog**
+It also supports a variety of inputs and outputs. Here mainly the built-in kubernetes support is missing.
+
+**Logs - Logstash**
+Popular project with a rich ecosystem with built-in kubernetes support. Developed under the Elastic umbrella and a clear addiction to use ElasticSearch as a backend.
+
+**Logs - Promtail**
+Is not vendor-neutral on the output side, mainly designed for the Loki backend
+
+**Logs - Flume**
+Seems to have no active community anymore
+
+**Traces - OpenCensus/Jaeger Agent**
+Both agents are deprecated in favor of using otel-collector.
+
+**Metrics - Prometheus**
+Prometheus is a collector and backend at the same time. While you could reduce the backend settings to a minimum and mainly use the metrics forwarding to act as a collector/forwarder, the solution will still be not flexible. Only prometheus compatible backends will be supported.
+
```

Comment:
The VictoriaMetrics agent might be worth to mention here. It is compatible with Prometheus (scraping, remote write), but has a smaller footprint. It can even deal with the Prometheus Operator CRDs.

Reply:
true, could you propose a section please

Reply:
lmk when it's ready for review again

Reply:
### Metrics - VictoriaMetrics Agent
The VictoriaMetrics project provides an agent that can receive metrics via different protocols and forward to a Prometheus compatible backend using the remote write protocol. The project claims to be more resource friendly than a full Prometheus instance. Their Kubernetes operator can consume Prometheus-Operator objects.